DataDog/datadog-operator
# Minimal dashboard

### What does this PR do?

Abillity to make a minimal datadog dashboard that is validated, periodically synced, and able to be updated and deleted. Every field other than widgets is configurable.

### Motivation
https://datadoghq.atlassian.net/jira/software/c/projects/CECO/boards/5438?selectedIssue=CECO-1352

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: vX.Y.Z
* Cluster Agent: vX.Y.Z

### Describe your test plan

Write there any instructions and details you may have to test your PR.
Using this manifest:
```
apiVersion: datadoghq.com/v1alpha1
kind: DatadogDashboard
metadata:
  name: example-dashboard1
  namespace: system 
spec:
  title: first_test
  layoutType: ordered
  tags:
    - "team:tagtest"
  reflowType: auto
  templateVariables:
    - availableValues: 
        value: 
          - host1
          - host2
      name: first
      prefix: 
        value: bar-foo
  templateVariablePresets:
    - name: saved_view1
      templateVariables:
        - name: foo-bar
          values:
            - foo
            - bar
  ```
 - Create a dashboard 
 - Check that the dashboard is created with the above configuration
 - Change the manifest and update dashboard -> there should be the same change in app
 - Change the force sync period to 30 seconds and change the configuration of the dashboard in app -> after ~30 seconds the dashboard should be reconciled and reverted back to the original
 - Delete the custom resource -> this should delete the dashboard in app as well. 
### Checklist
